src/vos/vos_iterator.c

```diff
@@ -660,7 +675,7 @@ vos_iterate_internal(vos_iter_param_t *param, vos_iter_type_t type,
 		if (rc != 0) {
 			if (vos_dtx_continue_detect(rc)) {
 				pre_cb = NULL;
-				post_cb = NULL;
+				post_cb = vos_iter_detect_dtx_cb;
```

Comment:
This is easier to see in this patch...wouldn't vos_iter_probe and vos_iter_next return -DER_INPROGRESS ?

Reply:
vos_dtx_continue_detect() is used for detecting whether there are MORE 'prepared' DTXs when we already hit the first 'prepared' DTX, if have, we may have chance to check with the leader together to improve the efficiency. But even if we miss to detect more, it can be found when we re-entry after resolving current known 'prepared' DTX status. Only the efficiency may be different, not affect the correctness.

Here, both ->probe() and ->next() may return -DER_INPROGRESS, if they are recursively called by other vos_iteration(), then the out level callers will check vos_dtx_continue_detect() and go ahead. If they are the first level iterations, then the caller will begin dtx_refresh() with related leader(s) and handle the other potential 'prepared' DTX when re-entry.

Reply:
As discussed in the transaction chat, I think this needs to change to probe/next.  If either of those return -DER_INPROGRESS, fetch will never happen.  Also, nested iterators are only used internally for rebuild/aggregation type stuff.

Reply:
Nevertheless, I'm ok with landing this since it's not a regression and fixing this issue later.